controllers/secretproviderclasspodstatus_controller.go

```diff
@@ -308,7 +350,11 @@ func (r *SecretProviderClassPodStatusReconciler) patchSecretWithOwnerRef(ctx con
 		Name:      name,
 	}
 	err := r.Client.Get(ctx, secretKey, secret)
-	if err != nil && !errors.IsNotFound(err) {
+	if apierrors.IsNotFound(err) {
```

Comment:
should we check for nil first?

Reply:
*logically* `apierrors.IsNotFound` returns false if error is nil or error is anything other than not found. So we can do the check even without the nil check first.

But I think the code might be more readable with the `err != nil` check. So I've updated the PR.